### DISCUSSION TITLE
Gurobi Search Path does not include current folder

### DISCUSSION THREAD

**Discussion by sfiruch** [google/or-tools#3774] [Feature requests]
**What version of OR-Tools and what language are you using?**
Version: 9.6.2534
Language: C#

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
Gurobi

**What operating system (Linux, Windows, ...) and version?**
Windows 11 (22621.1555)

**What did you do?**
1. Place `gurobi100.dll`, `gurobi100.NET.dll` and `gurobi.lic` in a folder
2. Compile and run the following C# program in the same folder:
```cs
using var m = Solver.CreateSolver("Gurobi");
```

**What did you expect to see**
No error message

**What did you see instead?**
```
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
W0000 00:00:1683214867.234411    6748 linear_solver.cc:644] Support for Gurobi not linked in, or the license was not found.
```

Only when I copy the libraries to the folder `C:\Program Files\gurobi1000\win64\bin`, the application works without error. Apparently, the application searches in these hardcoded paths for the Gurobi libraries, but not the current directory:
![image](https://user-images.githubusercontent.com/4119188/236213278-3968c4de-4356-4ab6-8ab8-bd7942fc308b.png)

**Anything else we should know about your project / environment**
It's common practice under Windows to colocate DLLs with the application executable.

**Comment by lperron** [comment]
You can pass the exact path of the library.

**Reply by sfiruch** [reply]
Correct. It caught me by surprise, because it doesn't behave like other Win32 libraries. The "normal" search order is documented here: https://learn.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order#standard-search-order-for-packaged-apps

**Comment by lperron** [comment]
On unix, it is not recommended to search on the current dir.
I need to patch gurobi/environment.cc

**Comment by lperron** [comment]
and pushed to main.

**Reply by sfiruch** [reply]
Thanks!!
